### PR TITLE
boards: arm: nrf5340dk_nrf5340: Fix missing pull ups on i2c

### LINKED ISSUE #53782: nrf5340dk: missing i2c bias pull up
**Describe the bug**
The nRF5340dk board does not have external i2c pull resistors, therefore `bias-pull-up;` needs to be added to allow i2c to work.

### PR DESCRIPTION
**Pull request by nordicjm**
Fixes an issue whereby the I2C lines are not pulled up with internal resistors.

Fixes #53782

### CODE REVIEW

**Review by gmarull — APPROVED** [review]

File: boards/arm/nrf5340dk_nrf5340/nrf5340_cpuapp_common-pinctrl.dtsi [hunk 1]
```diff
@@ -8,6 +8,7 @@
 		group1 {
 			psels = <NRF_PSEL(TWIM_SDA, 1, 2)>,
 				<NRF_PSEL(TWIM_SCL, 1, 3)>;
+			bias-pull-up;
```
Comment (anangl) [inline boards/arm/nrf5340dk_nrf5340/nrf5340_cpuapp_common-pinctrl.dtsi:11]: This should not be added at the board level. If a shield needs to use SoC internal pull-ups, then it should add them in its overlay. Normally, external pull-ups should be used for I2C bus.

**Review by anangl — CHANGES_REQUESTED** [review]

File: boards/arm/nrf5340dk_nrf5340/nrf5340_cpuapp_common-pinctrl.dtsi (hunk 1 above)
Comment (nordicjm) [reply boards/arm/nrf5340dk_nrf5340/nrf5340_cpuapp_common-pinctrl.dtsi:11]: Boards like nrf52840dk have external pull up resistors, I did not see any pull up resistors on the nrf5340dk, hence why I added them here

Comment (anangl) [reply boards/arm/nrf5340dk_nrf5340/nrf5340_cpuapp_common-pinctrl.dtsi:11]: Those are present also on the nRF5340 DK, see [Signal switches](https://infocenter.nordicsemi.com/topic/ug_nrf5340_dk/UG/dk/signal_switches.html) (U6 specifically).

Comment (nordicjm) [reply boards/arm/nrf5340dk_nrf5340/nrf5340_cpuapp_common-pinctrl.dtsi:11]: Not sure how I missed that, right you are.

**Pull request closed without merging**